From 39b49afbe75076d73903ae4610305d8080f3ffba Mon Sep 17 00:00:00 2001
From: weiweij <weiweij@motorola.com>
Date: Fri, 8 Jan 2016 09:08:12 +0800
Subject: [PATCH 600/959] IKSWM-15104: arm/dts: integrate battery data of GA40

Integrating newest battery profile data

Change-Id: I5e59a4bcf987a983311103289681361d9f7d4781
Signed-off-by: weiweij <weiweij@motorola.com>
Reviewed-on: http://gerrit.mot.com/810574
SLTApproved: Slta Waiver <sltawvr@motorola.com>
SME-Granted: SME Approvals Granted
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Kenneth Kessler <kennykessler@motorola.com>
Reviewed-by: Jason Knopsnyder <jasonk@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 .../boot/dts/qcom/batterydata-ga40-3000mah.dtsi    | 60 ++++++++++++++++++++++
 arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi    |  8 +--
 2 files changed, 61 insertions(+), 7 deletions(-)
 create mode 100644 arch/arm/boot/dts/qcom/batterydata-ga40-3000mah.dtsi

diff --git a/arch/arm/boot/dts/qcom/batterydata-ga40-3000mah.dtsi b/arch/arm/boot/dts/qcom/batterydata-ga40-3000mah.dtsi
new file mode 100644
index 00000000000..5e6c4f935c1
--- /dev/null
+++ b/arch/arm/boot/dts/qcom/batterydata-ga40-3000mah.dtsi
@@ -0,0 +1,60 @@
+/* Copyright (c) 2016, The Linux Foundation. All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 and
+ * only version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+qcom,ga40-3000mah {
+	/* #2265165_Moto_Athene_GA40_3000mAh_averaged_MasterSlave_Dec18th2015*/
+	qcom,max-voltage-uv = <4350000>;
+	qcom,nom-batt-capacity-mah = <3000>;
+	qcom,batt-id-kohm = <243>;
+	qcom,battery-beta = <3900>;
+	qcom,battery-type = "GA40-3000mAh";
+	qcom,chg-rslow-comp-c1 = <2456357>;
+	qcom,chg-rslow-comp-c2 = <2731891>;
+	qcom,chg-rs-to-rslow = <746576>;
+	qcom,chg-rslow-comp-thr = <0x99>;
+	qcom,checksum = <0xA2E0>;
+	qcom,gui-version = "PMI8950GUI - 2.0.0.15";
+	qcom,fg-profile-data = [
+		DA 83 4D 7C
+		67 80 F0 75
+		3C 83 57 6A
+		5D 78 B6 84
+		1B 82 43 99
+		A3 B7 AE C3
+		58 0F AB 82
+		D5 79 D0 7C
+		F7 71 23 83
+		30 7A 4D 87
+		E7 82 22 82
+		C9 99 E5 BC
+		F6 C9 83 0D
+		D5 0B 34 60
+		CE 6E 64 FC
+		2C 0B 2D 46
+		93 41 00 00
+		CE 46 6E 33
+		FE 2A 00 00
+		00 00 00 00
+		00 00 00 00
+		66 6B 0F 6A
+		A9 75 2E 81
+		B7 7C AF 70
+		44 74 78 80
+		72 7C 34 70
+		94 6E BB A8
+		1D 27 59 2C
+		4E A0 71 0C
+		28 00 FF 36
+		F0 11 30 03
+		00 00 00 0C
+	];
+};
diff --git a/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi b/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
index 114cda600c5..3e31ebb589b 100644
--- a/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
@@ -441,13 +441,7 @@
 /{
 	mot_batterydata: qcom,battery-data {
 		qcom,batt-id-range-pct = <100>;
-		#include "batterydata-fx30-lg.dtsi"
-	};
-};
-
-&mot_batterydata {
-	qcom,fx30-lg {
-		qcom,max-voltage-uv = <435000>;
+		#include "batterydata-ga40-3000mah.dtsi"
 	};
 };
 
-- 
2.11.0

